src/main/java/org/spongepowered/api/event/keyboard/PlayerChangeKeyCodeEvent.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.keyboard;
+
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.keyboard.KeyBinding;
+import org.spongepowered.api.keyboard.KeyCode;
+
+import java.util.Optional;
+
+/**
+ * Is fired when a {@link Player} the {@link KeyCode}
+ * of a specific {@link KeyBinding} modifies.
+ */
+public interface PlayerChangeKeyCodeEvent extends PlayerKeyboardEvent {
```

Comment:
@Mumfrey 

In the past I would have quite disagreed with you but this is a case where commands were meant for...since you can easily register a command and have the client send up that command.

I mean I do this currently with Backpack for Forge... Press B, send up /backpack.